diff --git a/linux-cachyos/PKGBUILD b/linux-cachyos/PKGBUILD
index d0a100ac..9127f4e8 100644
--- a/linux-cachyos/PKGBUILD
+++ b/linux-cachyos/PKGBUILD
@@ -149,6 +149,9 @@
 # Enable this after the profiles have been generated
 : "${_propeller_profiles:=no}"
 
+# Include config overrides
+source "${PWD}/kernel-configs/PKGBUILD.overrides"
+
 # ATTENTION: Do not modify after this line
 _is_lto_kernel() {
     [[ "$_use_llvm_lto" = "thin" || "$_use_llvm_lto" = "full"  || "$_use_llvm_lto" = "thin-dist" ]]
@@ -212,9 +215,13 @@ source=(
     "https://github.com/CachyOS/linux/releases/download/${_srctag}/${_srctag}.tar.gz"
     "config")
 
+source+=("file://${PWD}/kernel-configs/0101-kernel-patches.patch"
+         "file://${PWD}/kernel-configs/config.kernel_extra")
+
 # LLVM makedepends
 if _is_lto_kernel; then
-    makedepends+=(clang llvm lld)
+    _llvm_pkgver=$(pacman -Si llvm | grep ^Version | head -1 | grep -o '[^ :][0-9].[^ -]*')
+    makedepends+=("clang=${_llvm_pkgver}" "lld=${_llvm_pkgver}" "llvm=${_llvm_pkgver}")
     source+=("${_patchsource}/misc/dkms-clang.patch")
     BUILD_FLAGS=(
         CC=clang
@@ -245,8 +252,8 @@ fi
 
 # Use generated AutoFDO Profile
 if [ "$_autofdo" = "yes" ] && [ -n "$_autofdo_profile_name" ]; then
-    if [ -e "$_autofdo_profile_name" ]; then
-        source+=("$_autofdo_profile_name")
+    if [ -e "${PWD}/kernel-configs/${_autofdo_profile_name}" ]; then
+        source+=("file://${PWD}/kernel-configs/${_autofdo_profile_name}")
     else
         _die "Failed to find file ${_autofdo_profile_name}"
     fi
@@ -254,8 +261,8 @@ fi
 
 # Use generated Propeller Profile
 if [ "$_propeller" = "yes" ] && [ "$_propeller_profiles" = "yes" ]; then
-    source+=(propeller_cc_profile.txt
-             propeller_ld_profile.txt)
+    source+=("file://${PWD}/kernel-configs/propeller_cc_profile.txt"
+             "file://${PWD}/kernel-configs/propeller_ld_profile.txt")
 fi
 
 if [ "$_build_r8125" = "yes" ]; then
@@ -279,6 +286,7 @@ export KBUILD_BUILD_USER="$pkgbase"
 export KBUILD_BUILD_TIMESTAMP="$(date -Ru${SOURCE_DATE_EPOCH:+d @$SOURCE_DATE_EPOCH})"
 
 _die() { error "$@" ; exit 1; }
+_make() { make KCFLAGS="-march=x86-64-v3 -mtune=raptorlake" "$@" ; }
 
 prepare() {
     cd "$_srcname"
@@ -300,7 +308,7 @@ prepare() {
     done
 
     echo "Setting config..."
-    cp ../config .config
+    cat ../config ../config.kernel_extra > .config
 
     ### Select CPU optimization
     if [ -n "$_processor_opt" ]; then
@@ -487,7 +495,7 @@ prepare() {
     if [ "$_localmodcfg" = "yes" ]; then
         if [ -e "$_localmodcfg_path" ]; then
             echo "Running Steven Rostedt's make localmodconfig now"
-            make "${BUILD_FLAGS[@]}" LSMOD="${_localmodcfg_path}" localmodconfig
+            _make "${BUILD_FLAGS[@]}" LSMOD="${_localmodcfg_path}" localmodconfig
         else
             _die "No modprobed.db data found"
         fi
@@ -495,19 +503,19 @@ prepare() {
 
     ### Rewrite configuration
     echo "Rewrite configuration..."
-    make "${BUILD_FLAGS[@]}" prepare
-    yes "" | make "${BUILD_FLAGS[@]}" config >/dev/null
+    _make "${BUILD_FLAGS[@]}" prepare
+    yes "" | _make "${BUILD_FLAGS[@]}" config >/dev/null
     diff -u ../config .config || :
 
     ### Prepared version
-    make -s kernelrelease > version
+    _make -s kernelrelease > version
     echo "Prepared $pkgbase version $(<version)"
 
     ### Running make nconfig
-    [ "$_makenconfig" = "yes" ] && make "${BUILD_FLAGS[@]}" nconfig
+    [ "$_makenconfig" = "yes" ] && _make "${BUILD_FLAGS[@]}" nconfig
 
     ### Running make xconfig
-    [ "$_makexconfig" = "yes" ] &&  make "${BUILD_FLAGS[@]}" xconfig
+    [ "$_makexconfig" = "yes" ] && _make "${BUILD_FLAGS[@]}" xconfig
 
     ### Save configuration for later reuse
     echo "Save configuration for later reuse..."
@@ -544,10 +552,10 @@ _sign_modules() {
 
 build() {
     cd "$_srcname"
-    make "${BUILD_FLAGS[@]}" -j"$(nproc)" all
+    _make "${BUILD_FLAGS[@]}" -j"$(nproc)" all
 
     if ! _is_ci_build; then
-        make -C tools/bpf/bpftool vmlinux.h feature-clang-bpf-co-re=1
+        _make -C tools/bpf/bpftool vmlinux.h feature-clang-bpf-co-re=1
     fi
 
     local MODULE_FLAGS=(
@@ -560,7 +568,7 @@ build() {
     if [ "$_build_nvidia_open" = "yes" ]; then
         cd "${srcdir}/${_nv_open_pkg}"
         MODULE_FLAGS+=(IGNORE_CC_MISMATCH=yes)
-        CFLAGS= CXXFLAGS= LDFLAGS= make "${BUILD_FLAGS[@]}" "${MODULE_FLAGS[@]}" -j"$(nproc)" modules
+        CFLAGS= CXXFLAGS= LDFLAGS= _make "${BUILD_FLAGS[@]}" "${MODULE_FLAGS[@]}" -j"$(nproc)" modules
     fi
 
     if [ "$_build_zfs" = "yes" ]; then
@@ -575,12 +583,12 @@ build() {
             --libdir=/usr/lib --datadir=/usr/share --includedir=/usr/include \
             --with-udevdir=/lib/udev --libexecdir=/usr/lib/zfs --with-config=kernel \
             --with-linux="${srcdir}/$_srcname"
-        make "${BUILD_FLAGS[@]}"
+        _make "${BUILD_FLAGS[@]}"
     fi
 
     if [ "$_build_r8125" = "yes" ]; then
         cd "${srcdir}/r8125"
-        make "${BUILD_FLAGS[@]}" KERNELDIR="$srcdir/$_srcname" modules
+        _make "${BUILD_FLAGS[@]}" KERNELDIR="$srcdir/$_srcname" modules
     fi
 
 }
@@ -606,13 +614,13 @@ _package() {
     echo "Installing boot image..."
     # systemd expects to find the kernel here to allow hibernation
     # https://github.com/systemd/systemd/commit/edda44605f06a41fb86b7ab8128dcf99161d2344
-    install -Dm644 "$(make -s image_name)" "$modulesdir/vmlinuz"
+    install -Dm644 "$(_make -s image_name)" "$modulesdir/vmlinuz"
 
     # Used by mkinitcpio to name the kernel
     echo "$pkgbase" | install -Dm644 /dev/stdin "$modulesdir/pkgbase"
 
     echo "Installing modules..."
-    ZSTD_CLEVEL=19 make "${BUILD_FLAGS[@]}" INSTALL_MOD_PATH="$pkgdir/usr" INSTALL_MOD_STRIP=1 \
+    ZSTD_CLEVEL=19 _make "${BUILD_FLAGS[@]}" INSTALL_MOD_PATH="$pkgdir/usr" INSTALL_MOD_STRIP=1 \
         DEPMOD=/doesnt/exist  modules_install  # Suppress depmod
 
     # remove build links
@@ -626,7 +634,7 @@ _package-headers() {
     if _is_lto_kernel; then
         provides+=(linux-cachyos-lto-headers=$_kernver)
         replaces=(linux-cachyos-lto-headers)
-        depends+=(clang llvm lld)
+        depends+=("clang=${_llvm_pkgver}" "lld=${_llvm_pkgver}" "llvm=${_llvm_pkgver}")
     fi
 
     cd "${_srcname}"
@@ -686,7 +694,7 @@ _package-headers() {
     fi
 
     echo "Installing unstripped VDSO..."
-    make INSTALL_MOD_PATH="$pkgdir/usr" vdso_install \
+    _make INSTALL_MOD_PATH="$pkgdir/usr" vdso_install \
       link=  # Suppress build-id symlinks
 
     echo "Removing unneeded architectures..."
@@ -795,12 +803,22 @@ _package-r8125() {
     echo "blacklist r8169" > "${pkgdir}/usr/lib/modprobe.d/${pkgname}.conf"
 }
 
+_package-signkey() {
+    pkgdesc="Signing key used for the ${pkgdesc} kernel"
+    depends=("${pkgbase}-headers=${pkgver}-${pkgrel}")
+
+    cd "${_srcname}"
+    local builddir="$pkgdir/usr/lib/modules/$(<version)/build"
+    install -Dt "${builddir}/certs" -m644 certs/signing_key.{pem,x509}
+}
+
 pkgname=("$pkgbase")
 [ "$_build_debug" = "yes" ] && pkgname+=("$pkgbase-dbg")
 pkgname+=("$pkgbase-headers")
 [ "$_build_zfs" = "yes" ] && pkgname+=("$pkgbase-zfs")
 [ "$_build_nvidia_open" = "yes" ] && pkgname+=("$pkgbase-nvidia-open")
 [ "$_build_r8125" = "yes" ] && pkgname+=("$pkgbase-r8125")
+pkgname+=("${pkgbase}-signkey")
 for _p in "${pkgname[@]}"; do
     eval "package_$_p() {
     $(declare -f "_package${_p#$pkgbase}")
@@ -810,4 +828,10 @@ done
 
 b2sums=('a0885a688e8ea83da755115a86926134f8795b93c7d2eb1c00baff521ed10ba0aeb2691ad26454da20f2cecb169b6d655705b7e11fa6d108428ca1d2c0e5d696'
         'b55556d1ebec83a529359f74e7231d48d85066be80a472591c3e8c8f258050ce3132e277e367f793d0d93896224ec4bd6e0ebf3fdb0ae674b23141d66802dc16'
-        'ea26c88950fc06b6ffab93b30e3beacc7d26571a70262334ca8b001dc7899bf96b47d703fbaa7f4e47765c3dafccc23c58a4d4da2169b8ee50012afcb7a1dd96')
+        '3e2b8f8bfbaf7acb823376b5dbded67dca84bb003ea84917e3ec55a03549764f698642fe7feb933c8e73917e284fca68c1676780bfff361dff0f2212281ca99a'
+        'd884b1cbf3aa605d6375c3fb8095b9a65bc28d9f7aef2088a45b76df7c5a393ad68767878bbf93b5c5ffd87063df10d670e1f145945c2d6c2132c8dc12d1f30c'
+        'ea26c88950fc06b6ffab93b30e3beacc7d26571a70262334ca8b001dc7899bf96b47d703fbaa7f4e47765c3dafccc23c58a4d4da2169b8ee50012afcb7a1dd96'
+        'b64931b5fcf47fab636232a94abcd107f4faae80cb9282e8aa5cfbb1a9fb113ef382843e4122225e735308f616e2cd4e7447ed1279eae2bd8f7705c19b7e4125'
+        '7237770166b77479fc932d22910787718439171c7f85129848069f63f032ce0c87246850a1422c19e80e8e8ab9267e9a26eb4ea2cd2dc74e543a509154a735af'
+        '489df32365e3e291ce7d331d818b1de564da4221cba6009dbe1d57caa1f0c5e950944a728bcfc408b18845e8033e9492692b6e83a9eeadf00262f6ebb662c708'
+        '226c64dd989ec0c4c444d048707e5d56be4a7ffa59ada31f197015c65a87e7935c8a0a1d6a9d35947e60f90505e5cffb3df9824aec71b2f188bcaa2e89403e0b')
