diff --git a/linux-cachyos/PKGBUILD b/linux-cachyos/PKGBUILD
index a0cc3d4b..bba78366 100644
--- a/linux-cachyos/PKGBUILD
+++ b/linux-cachyos/PKGBUILD
@@ -154,6 +154,9 @@
 # Enable this after the profiles have been generated
 : "${_propeller_profiles:=no}"
 
+# Include config overrides
+source "${PWD}/kernel-configs/PKGBUILD.overrides"
+
 # ATTENTION: Do not modify after this line
 _is_lto_kernel() {
     [[ "$_use_llvm_lto" = "thin" || "$_use_llvm_lto" = "full"  || "$_use_llvm_lto" = "thin-dist" ]]
@@ -180,7 +183,7 @@ _minor=3
 #_rcver=rc8
 pkgver=${_major}.${_minor}
 _tagrel=1
-pkgrel=1
+pkgrel=3
 _srcname=cachyos-${_major}.${_minor}-${_tagrel}
 pkgdesc='Linux EEVDF + LTO + AutoFDO + Propeller Cachy Sauce Kernel by CachyOS with other patches and improvements.'
 _kernver="$pkgver-$pkgrel"
@@ -219,9 +222,13 @@ source=(
     "https://github.com/CachyOS/linux/releases/download/${_srcname}/${_srcname}.tar.gz"
     "config")
 
+source+=("file://${PWD}/kernel-configs/0101-kernel-patches.patch"
+         "file://${PWD}/kernel-configs/config.kernel_extra")
+
 # LLVM makedepends
 if _is_lto_kernel; then
-    makedepends+=(clang llvm lld)
+    _llvm_pkgver=$(pacman -Si llvm | grep ^Version | head -1 | grep -o '[^ :][0-9].[^ -]*')
+    makedepends+=("clang=${_llvm_pkgver}" "lld=${_llvm_pkgver}" "llvm=${_llvm_pkgver}")
     source+=("${_patchsource}/misc/dkms-clang.patch")
     BUILD_FLAGS=(
         CC=clang
@@ -253,8 +260,8 @@ fi
 
 # Use generated AutoFDO Profile
 if [ "$_autofdo" = "yes" ] && [ -n "$_autofdo_profile_name" ]; then
-    if [ -e "$_autofdo_profile_name" ]; then
-        source+=("$_autofdo_profile_name")
+    if [ -e "${PWD}/kernel-configs/${_autofdo_profile_name}" ]; then
+        source+=("file://${PWD}/kernel-configs/${_autofdo_profile_name}")
     else
         _die "Failed to find file ${_autofdo_profile_name}"
     fi
@@ -262,8 +269,8 @@ fi
 
 # Use generated Propeller Profile
 if [ "$_propeller" = "yes" ] && [ "$_propeller_profiles" = "yes" ]; then
-    source+=(propeller_cc_profile.txt
-             propeller_ld_profile.txt)
+    source+=("file://${PWD}/kernel-configs/propeller_cc_profile.txt"
+             "file://${PWD}/kernel-configs/propeller_ld_profile.txt")
 fi
 
 if [ "$_build_r8125" = "yes" ]; then
@@ -272,7 +279,7 @@ fi
 
 ## List of CachyOS schedulers
 case "$_cpusched" in
-    bore|rt-bore|hardened) # CachyOS Scheduler (BORE)
+    bore|rt-bore|hardened)
         source+=("${_patchsource}/sched/0001-bore-cachy.patch");;&
     bmq) ## Project C Scheduler
         source+=("${_patchsource}/sched/0001-prjc-cachy.patch");;
@@ -287,6 +294,7 @@ export KBUILD_BUILD_USER="$pkgbase"
 export KBUILD_BUILD_TIMESTAMP="$(date -Ru${SOURCE_DATE_EPOCH:+d @$SOURCE_DATE_EPOCH})"
 
 _die() { error "$@" ; exit 1; }
+_make() { make KCFLAGS="-march=x86-64-v3 -mtune=raptorlake" "$@" ; }
 
 prepare() {
     cd "$_srcname"
@@ -294,6 +302,7 @@ prepare() {
     echo "Setting version..."
     echo "-$pkgrel" > localversion.10-pkgrel
     echo "${pkgbase#linux}" > localversion.20-pkgname
+    echo "-exp" > localversion.99-exp
 
     local src
     for patch in "${source[@]}"; do
@@ -310,7 +319,7 @@ prepare() {
     done
 
     echo "Setting config..."
-    cp ../config .config
+    cat ../config ../config.kernel_extra > .config
 
     ### Select CPU optimization
     if [ -n "$_processor_opt" ]; then
@@ -334,9 +343,9 @@ prepare() {
 
     ### Selecting the CPU scheduler
     case "$_cpusched" in
-        cachyos|bore|hardened) scripts/config -e SCHED_BORE;;
+        bore|hardened) scripts/config -e SCHED_BORE;;
         bmq) scripts/config -e SCHED_ALT -e SCHED_BMQ;;
-        eevdf) ;;
+        cachyos|eevdf) ;;
         rt) scripts/config -e PREEMPT_RT;;
         rt-bore) scripts/config -e SCHED_BORE -e PREEMPT_RT;;
         *) _die "The value $_cpusched is invalid. Choose the correct one again.";;
@@ -492,7 +501,7 @@ prepare() {
     if [ "$_localmodcfg" = "yes" ]; then
         if [ -e "$_localmodcfg_path" ]; then
             echo "Running Steven Rostedt's make localmodconfig now"
-            make "${BUILD_FLAGS[@]}" LSMOD="${_localmodcfg_path}" localmodconfig
+            _make "${BUILD_FLAGS[@]}" LSMOD="${_localmodcfg_path}" localmodconfig
         else
             _die "No modprobed.db data found"
         fi
@@ -500,19 +509,19 @@ prepare() {
 
     ### Rewrite configuration
     echo "Rewrite configuration..."
-    make "${BUILD_FLAGS[@]}" prepare
-    yes "" | make "${BUILD_FLAGS[@]}" config >/dev/null
+    _make "${BUILD_FLAGS[@]}" prepare
+    yes "" | _make "${BUILD_FLAGS[@]}" config >/dev/null
     diff -u ../config .config || :
 
     ### Prepared version
-    make -s kernelrelease > version
+    _make -s kernelrelease > version
     echo "Prepared $pkgbase version $(<version)"
 
     ### Running make nconfig
-    [ "$_makenconfig" = "yes" ] && make "${BUILD_FLAGS[@]}" nconfig
+    [ "$_makenconfig" = "yes" ] && _make "${BUILD_FLAGS[@]}" nconfig
 
     ### Running make xconfig
-    [ "$_makexconfig" = "yes" ] &&  make "${BUILD_FLAGS[@]}" xconfig
+    [ "$_makexconfig" = "yes" ] && _make "${BUILD_FLAGS[@]}" xconfig
 
     ### Save configuration for later reuse
     echo "Save configuration for later reuse..."
@@ -543,10 +552,10 @@ _sign_modules() {
 
 build() {
     cd "$_srcname"
-    make "${BUILD_FLAGS[@]}" -j"$(nproc)" all
+    _make "${BUILD_FLAGS[@]}" -j"$(nproc)" all
 
     if ! _is_ci_build; then
-        make -C tools/bpf/bpftool vmlinux.h feature-clang-bpf-co-re=1
+        _make -C tools/bpf/bpftool vmlinux.h feature-clang-bpf-co-re=1
     fi
 
     local MODULE_FLAGS=(
@@ -559,7 +568,7 @@ build() {
     if [ "$_build_nvidia_open" = "yes" ]; then
         cd "${srcdir}/${_nv_open_pkg}"
         MODULE_FLAGS+=(IGNORE_CC_MISMATCH=yes)
-        CFLAGS= CXXFLAGS= LDFLAGS= make "${BUILD_FLAGS[@]}" "${MODULE_FLAGS[@]}" -j"$(nproc)" modules
+        CFLAGS= CXXFLAGS= LDFLAGS= _make "${BUILD_FLAGS[@]}" "${MODULE_FLAGS[@]}" -j"$(nproc)" modules
     fi
 
     if [ "$_build_zfs" = "yes" ]; then
@@ -574,12 +583,12 @@ build() {
             --libdir=/usr/lib --datadir=/usr/share --includedir=/usr/include \
             --with-udevdir=/lib/udev --libexecdir=/usr/lib/zfs --with-config=kernel \
             --with-linux="${srcdir}/$_srcname"
-        make "${BUILD_FLAGS[@]}"
+        _make "${BUILD_FLAGS[@]}"
     fi
 
     if [ "$_build_r8125" = "yes" ]; then
         cd "${srcdir}/r8125"
-        make "${BUILD_FLAGS[@]}" KERNELDIR="$srcdir/$_srcname" modules
+        _make "${BUILD_FLAGS[@]}" KERNELDIR="$srcdir/$_srcname" modules
     fi
 
 }
@@ -605,13 +614,13 @@ _package() {
     echo "Installing boot image..."
     # systemd expects to find the kernel here to allow hibernation
     # https://github.com/systemd/systemd/commit/edda44605f06a41fb86b7ab8128dcf99161d2344
-    install -Dm644 "$(make -s image_name)" "$modulesdir/vmlinuz"
+    install -Dm644 "$(_make -s image_name)" "$modulesdir/vmlinuz"
 
     # Used by mkinitcpio to name the kernel
     echo "$pkgbase" | install -Dm644 /dev/stdin "$modulesdir/pkgbase"
 
     echo "Installing modules..."
-    ZSTD_CLEVEL=19 make "${BUILD_FLAGS[@]}" INSTALL_MOD_PATH="$pkgdir/usr" INSTALL_MOD_STRIP=1 \
+    ZSTD_CLEVEL=19 _make "${BUILD_FLAGS[@]}" INSTALL_MOD_PATH="$pkgdir/usr" INSTALL_MOD_STRIP=1 \
         DEPMOD=/doesnt/exist  modules_install  # Suppress depmod
 
     # remove build links
@@ -635,7 +644,7 @@ _package-headers() {
     if _is_lto_kernel; then
         provides+=(linux-cachyos-lto-headers=$_kernver)
         replaces=(linux-cachyos-lto-headers)
-        depends+=(clang llvm lld)
+        depends+=("clang=${_llvm_pkgver}" "lld=${_llvm_pkgver}" "llvm=${_llvm_pkgver}")
     fi
 
     cd "${_srcname}"
@@ -695,7 +704,7 @@ _package-headers() {
     fi
 
     echo "Installing unstripped VDSO..."
-    make INSTALL_MOD_PATH="$pkgdir/usr" vdso_install \
+    _make INSTALL_MOD_PATH="$pkgdir/usr" vdso_install \
       link=  # Suppress build-id symlinks
 
     echo "Removing unneeded architectures..."
@@ -804,12 +813,22 @@ _package-r8125() {
     echo "install r8169 /usr/bin/modprobe r8125 || /usr/bin/modprobe --ignore-install r8169" > "${pkgdir}/usr/lib/modprobe.d/${pkgname}.conf"
 }
 
+_package-signkey() {
+    pkgdesc="Signing key used for the ${pkgdesc} kernel"
+    depends=("${pkgbase}-headers=${pkgver}-${pkgrel}")
+
+    cd "${_srcname}"
+    local builddir="$pkgdir/usr/lib/modules/$(<version)/build"
+    install -Dt "${builddir}/certs" -m644 certs/signing_key.{pem,x509}
+}
+
 pkgname=("$pkgbase")
 [ "$_build_debug" = "yes" ] && pkgname+=("$pkgbase-dbg")
 pkgname+=("$pkgbase-headers")
 [ "$_build_zfs" = "yes" ] && pkgname+=("$pkgbase-zfs")
 [ "$_build_nvidia_open" = "yes" ] && pkgname+=("$pkgbase-nvidia-open")
 [ "$_build_r8125" = "yes" ] && pkgname+=("$pkgbase-r8125")
+pkgname+=("${pkgbase}-signkey")
 for _p in "${pkgname[@]}"; do
     eval "package_$_p() {
     $(declare -f "_package${_p#$pkgbase}")
@@ -818,5 +837,11 @@ for _p in "${pkgname[@]}"; do
 done
 
 b2sums=('d6ab08acd91f405f36205070cdebab99d244e439ca00d823c4068aeba583818dd3c1fdc0d0b3836406ecb1a93b611c637533b50ae96e9bda3153682bbb9f33bd'
-        'bd404bba9f1d30d7a82e7d147e11d34e28a75d00f511e1e5cbfea51a33f814eb492c28a4f0bda87546d5c81b980dce69bdd7ac5f02860f4d3c003b2073f2aa9f'
-        'c992567bd7dd8553432be496ffa1c17e2f5ebe9c7edb51945cf977e1b742dd6517c210d8843bb82744ca705efd07f8027cd7dde41b50215ebd707a34aa81462e')
+        '7bb5113dbc67e8e2ce5c5473ae1b08973af5adba0a6a14c64a213bb116e5a172d40b7c274b85ad15553511484ee1f120e0372251e242c6f87ce6920235f0c136'
+        'cc7334bd2c995ad41beb2b3f8c0accda9a31383981dfa9400a1eddbc78b45e667a08bec0ce7edb7d4333170e7b7c62edb4c5957bc847c51f87c220556a99ec87'
+        '456ac00a11e65ce06c01c2ddf9b2cb3949a4fe71d1dfdfe9a34f9cf65a1851f01fe72c62089b25aad13c4540127d856c6412b5e8e38dfc2d8916e8f560dda873'
+        'c992567bd7dd8553432be496ffa1c17e2f5ebe9c7edb51945cf977e1b742dd6517c210d8843bb82744ca705efd07f8027cd7dde41b50215ebd707a34aa81462e'
+        '22009b50a3ee7826db4757d316fc3b4401ac60c7825eb540c53b95922fe9cf95566de4cd0c847f6e4444e08e8910b17c30c943348fdfc6a841b33da16d5f572c'
+        '5108a1b3d150cd5768d810175e115ad5082b83ccf575aef4368557e9d41a79d16a739da340df76c38c2b3627960605af9e9d008c3d8f2ece6f0a4f9d7df8f1fc'
+        '69788445ccc0ec5f6d647b8955e22d8b7136931aae22e47dca2d97f50dca355ac13d34ed7f8ea6cd38e915dee1acfd0fb7438a3eac597824d147d98915730abc'
+        '9dc1a5a46d8ecf606323926f22b4ce0aaf910dc47fd9ab9b8d08d1600e0bb45109babf7098f390562d8d8456239bb44b7db13b175fe2f529b9784a603dc11fbe')
